hive-metastore/src/main/java/org/apache/iceberg/hive/HiveTableOperations.java

```diff
@@ -437,6 +451,18 @@ private Table newHmsTable(TableMetadata metadata) {
             null,
             TableType.EXTERNAL_TABLE.toString());
 
+    // hive2 does not support setting table ownership type, therefore we leave it as
+    // a table property. Once upgraded to hive3, then we can simply set
+    // newTable.setOwnerType(PrincipalType.valueOf(metadata.get(HMS_TABLE_OWNER_TYPE)))
+    String ownerType = metadata.property(TableProperties.HMS_TABLE_OWNER_TYPE, null);
+    if (ownerType != null && ownerType.equalsIgnoreCase(PrincipalType.GROUP.name())) {
+      newTable
+          .getParameters()
+          .put(TableProperties.HMS_TABLE_OWNER_TYPE, PrincipalType.GROUP.name());
```

Comment:
Cant you simply put ownerType instead of PrincipalType.GROUP.name()?

Reply:
sure, I was originally think about if I simply put a string like `"GROUP"` down, the later readers will wonder why it's not `"Group"`, `"group"` or `"groups"`. This would demonstrate where the string value comes from. But yes, I'm good with just a plain string here.

Reply:
nvm, ignore my previous comment. I think I mis-understood what you are saying. Yes, just directly put `ownerType` works. I probably shouldn't use `equalsIgnoreCase` in L458 in the first place. I think we should make clear to users, if they want to tweak owner type, their key should be all-cap `"GROUP"` in the property.

Reply:
Decided to not support group ownership for tables here since hive2 does not support it. Hopefully this will simplify this PR. So this thread may not be relevant anymore.